src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -327,23 +327,24 @@ void AddAssemblies (ZipArchiveEx apk, bool debug, bool compress, IDictionary<str
 				sourcePath = CompressAssembly (assembly);
 
 				// Add assembly
-				AddFileToArchiveIfNewer (apk, sourcePath, GetTargetDirectory (assembly.ItemSpec) + "/"  + Path.GetFileName (assembly.ItemSpec), compressionMethod: UncompressedMethod);
+				var assemblyPath = GetAssemblyPath (assembly, frameworkAssembly: false);
+				AddFileToArchiveIfNewer (apk, sourcePath, assemblyPath + Path.GetFileName (assembly.ItemSpec), compressionMethod: UncompressedMethod);
 
 				// Try to add config if exists
 				var config = Path.ChangeExtension (assembly.ItemSpec, "dll.config");
-				AddAssemblyConfigEntry (apk, config);
+				AddAssemblyConfigEntry (apk, assemblyPath, config);
 
 				// Try to add symbols if Debug
 				if (debug) {
 					var symbols = Path.ChangeExtension (assembly.ItemSpec, "dll.mdb");
 
 					if (File.Exists (symbols))
-						AddFileToArchiveIfNewer (apk, symbols, AssembliesPath + Path.GetFileName (symbols), compressionMethod: UncompressedMethod);
+						AddFileToArchiveIfNewer (apk, symbols, assemblyPath + Path.GetFileName (symbols), compressionMethod: UncompressedMethod);
```

Comment:
Instead of `assemblyPath +`, should this instead be `Path.Combine()`?

Reply:
This is the path within the zip file, so we always want to use `/`.